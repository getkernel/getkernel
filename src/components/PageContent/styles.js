/**
 * PageContent styles.
 */
export default (theme) => ({
  root: {
    backgroundColor: '#eee',
    padding: theme.spacing(3),
  },
});
