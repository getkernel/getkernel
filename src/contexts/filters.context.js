import React, { createContext } from 'react';
import { filtersReducer } from '../reducers';
import { useLocalStorageReducer } from '../hooks';
import defaultState from '../reducers/filters.defaultState';

export const FiltersContext = createContext();
export const FiltersDispatchContext = createContext();

export function FiltersProvider({ children }) {
  const [state, dispatch] = useLocalStorageReducer(
    'filters',
    defaultState,
    filtersReducer
  );

  return (
    <FiltersContext.Provider value={state}>
      <FiltersDispatchContext.Provider value={dispatch}>
        {children}
      </FiltersDispatchContext.Provider>
    </FiltersContext.Provider>
  );
}

export const withFiltersProvider = (Component) => (props) => (
  <FiltersProvider>
    <Component {...props} />
  </FiltersProvider>
);
