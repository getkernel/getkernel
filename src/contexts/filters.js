/**
 * Filters context.
 */
import React, { createContext } from 'react';
import PropTypes from 'prop-types';
import { useLocalStorageReducer } from '../hooks';
import filtersReducer, { defaultState } from '../reducers/filters';

export const FiltersContext = createContext();
export const FiltersDispatchContext = createContext();

export const FiltersProvider = ({ children }) => {
  const [state, dispatch] = useLocalStorageReducer(
    'filters',
    defaultState,
    filtersReducer,
  );

  return (
    <FiltersDispatchContext.Provider value={dispatch}>
      <FiltersContext.Provider value={state}>
        {children}
      </FiltersContext.Provider>
    </FiltersDispatchContext.Provider>
  );
};

FiltersProvider.propTypes = {
  children: PropTypes.object.isRequired,
};
