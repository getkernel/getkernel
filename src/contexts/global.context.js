import React, { createContext } from 'react';
import { globalReducer } from '../reducers';
import { useLocalStorageReducer } from '../hooks';
import defaultState from '../reducers/global.defaultState';

export const GlobalContext = createContext();
export const GlobalDispatchContext = createContext();

export function GlobalProvider({ children }) {
  const [state, dispatch] = useLocalStorageReducer(
    'global',
    defaultState,
    globalReducer
  );

  return (
    <GlobalContext.Provider value={state}>
      <GlobalDispatchContext.Provider value={dispatch}>
        {children}
      </GlobalDispatchContext.Provider>
    </GlobalContext.Provider>
  );
}

export const withGlobalProvider = (Component) => (props) => (
  <GlobalProvider>
    <Component {...props} />
  </GlobalProvider>
);
