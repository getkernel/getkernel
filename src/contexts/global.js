/**
 * Global context.
 */
import React, { createContext } from 'react';
import PropTypes from 'prop-types';
import { useLocalStorageReducer } from '../hooks';
import globalReducer, { defaultState } from '../reducers/global';

export const GlobalContext = createContext();
export const GlobalDispatchContext = createContext();

export const GlobalProvider = ({ children }) => {
  const [state, dispatch] = useLocalStorageReducer(
    'global',
    defaultState,
    globalReducer,
  );

  return (
    <GlobalContext.Provider value={state}>
      <GlobalDispatchContext.Provider value={dispatch}>
        {children}
      </GlobalDispatchContext.Provider>
    </GlobalContext.Provider>
  );
};

GlobalProvider.propTypes = {
  children: PropTypes.object.isRequired,
};
