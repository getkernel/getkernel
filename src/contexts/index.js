import {
  KernelsContext,
  DispatchContext,
  KernelsProvider,
  withKernelsProvider,
} from './kernels.context';

export {
  KernelsContext,
  DispatchContext,
  KernelsProvider,
  withKernelsProvider,
};
