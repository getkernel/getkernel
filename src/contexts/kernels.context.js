import React, { createContext } from 'react';
import { kernelsReducer } from '../reducers';
import { useLocalStorageReducer } from '../hooks';
import defaultState from '../reducers/kernels.defaultState';

export const KernelsContext = createContext();
export const KernelsDispatchContext = createContext();

export function KernelsProvider({ children }) {
  const [state, dispatch] = useLocalStorageReducer(
    'kernels',
    defaultState,
    kernelsReducer
  );

  return (
    <KernelsContext.Provider value={state}>
      <KernelsDispatchContext.Provider value={dispatch}>
        {children}
      </KernelsDispatchContext.Provider>
    </KernelsContext.Provider>
  );
}
