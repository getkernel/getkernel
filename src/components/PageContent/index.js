/**
 * PageContent component.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Fade from '@material-ui/core/Fade';
import styles from './styles';

const useStyles = makeStyles(styles);

const PageContent = ({ children }) => {
  const classes = useStyles();

  return (
    <Fade in timeout={250}>
      <div className={classes.root}>{children}</div>
    </Fade>
  );
};

PageContent.propTypes = {
  children: PropTypes.node.isRequired,
};

export default PageContent;
