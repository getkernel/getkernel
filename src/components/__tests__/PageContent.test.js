import React, { useReducer } from 'react';
import { mount } from 'enzyme';
import { renderHook } from '@testing-library/react-hooks';
import { GlobalContext, GlobalProvider } from '../../contexts';
import PageContent from '../PageContent';
import globalReducer from '../../reducers/global';

describe('<PageContent />', () => {
  test('should not render children if isLoading is set to true', () => {
    const TestComponent = () => (
      <GlobalProvider>
        <PageContent>
          <h1>Hello Jest!</h1>
        </PageContent>
      </GlobalProvider>
    );
    const wrapper = mount(<TestComponent />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(PageContent).find('h1').length).toBe(0);
  });

  test('should render children if isLoading is set to false', () => {
    const testState = { isLoading: false };

    const { result } = renderHook(() => {
      return useReducer(globalReducer, testState);
    });

    const [state] = result.current;

    const TestComponent = () => (
      <GlobalContext.Provider value={state}>
        <PageContent>
          <h1>Hello Jest!</h1>
        </PageContent>
      </GlobalContext.Provider>
    );
    const wrapper = mount(<TestComponent />);
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find(PageContent)
        .find('h1')
        .text(),
    ).toBe('Hello Jest!');
  });
});
