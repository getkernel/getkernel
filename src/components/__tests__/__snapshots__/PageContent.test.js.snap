// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContent /> should correctly render PageContent component 1`] = `
<ForwardRef(Fade)
  in={true}
  timeout={250}
>
  <div
    className="makeStyles-root-1"
  >
    <h1>
      Hello Jest!
    </h1>
  </div>
</ForwardRef(Fade)>
`;
